tools/ctor_evaller.py

```diff
@@ -256,18 +264,17 @@ def read_and_delete(filename):
     try:
       shared.jsrun.timeout_run(proc, timeout=10, full_output=True, throw_on_failure=False)
     except Exception as e:
-      if 'Timed out' not in str(e): raise e
-      shared.logging.debug('ctors timed out\n')
+      if 'Timed out' not in str(e):
+        raise
       return (0, 0, 0, 0)
-    finally:
+    if shared.WINDOWS:
       time.sleep(0.5) # On Windows, there is some kind of race condition with Popen output stream related functions, where file handles are still in use a short period after the process has finished.
-      out_file_handle.close()
-      err_file_handle.close()
-      out_result = read_and_delete(out_file)
-      err_result = read_and_delete(err_file)
+    out_file_handle.close()
+    err_file_handle.close()
+    out_result = read_and_delete(out_file)
+    err_result = read_and_delete(err_file)
     if proc.returncode != 0:
-      shared.logging.debug('unexpected error while trying to eval ctors:\n' + out_result + '\n' + err_result)
-      return (0, 0, 0, 0)
+      shared.exit_with_error('unexpected error while trying to eval ctors:\n' + out_result + '\n' + err_result)
```

Comment:
I think this was just a warning during development of the feature, sounds good to change this and things like it to proper errors. Hopefully nothing will break on CI.